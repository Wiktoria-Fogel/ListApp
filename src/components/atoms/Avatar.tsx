import React from 'react';
import styled from 'styled-components/native';

type AvatarProps = {
  source: string;
};

export const Avatar: React.FC<AvatarProps> = ({source}) => {
  return (
    <Wrapper>
      <AvatarImage source={{uri: source}} />
    </Wrapper>
  );
};

const Wrapper = styled.View`
  height: 60px;
  width: 60px;
  margin-top: 8px;
`;

const AvatarImage = styled.Image`
  height: 100%;
  width: 100%;
  border-radius: 30px;
`;
