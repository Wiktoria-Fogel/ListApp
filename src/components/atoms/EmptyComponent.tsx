import React from 'react';
import styled from 'styled-components/native';
import {COLORS, VerticalSpacer} from '../theme';

interface EmptyComponentProps {
  title: string;
}

export const EmptyComponent: React.FC<EmptyComponentProps> = ({title}) => {
  return (
    <Wrapper>
      <VerticalSpacer height={12} />
      <Title>{title}</Title>
    </Wrapper>
  );
};

const Wrapper = styled.View`
  height: 100%;
  width: 100%;
  align-items: center;
`;

const Title = styled.Text`
  font-size: 18px;
  font-family: 'Raleway-Bold';
  color: ${COLORS.white};
`;
