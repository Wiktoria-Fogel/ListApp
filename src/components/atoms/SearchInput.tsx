import {TextInputProps, TouchableOpacity} from 'react-native';
import React, {useState} from 'react';
import styled from 'styled-components/native';
import {CancelIcon, MagnifierIcon} from '../../assets/svg/Icons';
import {COLORS} from '../theme';

type SearchInputProps = {
  onClear: () => void;
};

export const SearchInput: React.FC<SearchInputProps & TextInputProps> = ({
  onClear,
  ...other
}) => {
  const [isFocused, setIsFocused] = useState(false);

  return (
    <Wrap isFocused={isFocused || !!other.value}>
      <MagnifierIcon fill={isFocused ? COLORS.borderFocus : COLORS.border} />
      <StyledTextInput
        returnKeyType="done"
        textAlignVertical="center"
        placeholderTextColor={COLORS.grey600}
        {...other}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
      />
      {other.value ? (
        <TouchableOpacity onPress={onClear}>
          <CancelIcon />
        </TouchableOpacity>
      ) : null}
    </Wrap>
  );
};

const Wrap = styled.View<{
  isFocused: boolean;
}>`
  flex-direction: row;
  align-items: center;
  background-color: #ffffff;
  border: 1px solid
    ${({isFocused}) => (isFocused ? COLORS.borderFocus : COLORS.border)};
  border-radius: 8px;
  width: 100%;

  height: 44px;
  padding-left: 12px;
  padding-right: 12px;
  align-items: center;
  justify-content: center;
  height: 50px;
`;

const StyledTextInput = styled.TextInput`
  font-weight: 400;
  color: ${COLORS.grey900};
  padding: 0;
  margin-left: 10px;
  flex: 1;
`;
