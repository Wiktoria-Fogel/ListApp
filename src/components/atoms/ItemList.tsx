import React, {memo} from 'react';
import styled from 'styled-components/native';
import {PublicEvent} from '../../api/models/PublicEvent';
import {COLORS, VerticalSpacer} from '../theme';
import {Avatar} from './Avatar';

interface ItemListProps {
  item: PublicEvent;
}

export const ItemList: React.FC<ItemListProps> = memo(({item}) => {
  return (
    <ItemWrapper key={item.id}>
      <Row>
        <Avatar source={item.actor.avatar_url} />
        <InformationWrapper>
          <Title>{item.actor.display_login}</Title>
          <VerticalSpacer height={4} />
          <Description>{item.repo.name}</Description>
        </InformationWrapper>
      </Row>
    </ItemWrapper>
  );
});

const Row = styled.View`
  flex-direction: row;
  height: 100%;
`;

const ItemWrapper = styled.View`
  padding: 4px 12px;
  height: 86px;
  flex: auto;
  width: 100%;
  border-radius: 12px;
  background-color: ${COLORS.grey800};
`;

const Title = styled.Text`
  font-size: 14px;
  font-family: 'Raleway-Bold';
  color: ${COLORS.white};
`;

const Description = styled.Text`
  font-size: 12px;
  font-family: 'Raleway-Medium';
  color: ${COLORS.white};
`;

const InformationWrapper = styled.View`
  padding: 8px 12px;
  flex: 1;
`;
