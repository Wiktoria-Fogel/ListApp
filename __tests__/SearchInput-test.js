import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {SearchInput} from '../src/components/atoms/SearchInput';

test('SearchInput renders correctly', async () => {
  const result = renderer
    .create(<SearchInput onClear={() => jest.fn} />)
    .toJSON();
  await act(async () => {
    expect(result).toMatchSnapshot();
    return;
  });
});
