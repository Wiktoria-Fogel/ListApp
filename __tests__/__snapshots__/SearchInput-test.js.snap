// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<View
  isFocused={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderColor": "#adb5bd",
        "borderRadius": 8,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingLeft": 12,
        "paddingRight": 12,
        "width": "100%",
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="none"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={20}
    vbWidth={20}
    width={20}
  >
    <RNSVGGroup
      fill={null}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGPath
        clipRule={0}
        d="M12.655 12.06h.659l3.533 3.55a.88.88 0 010 1.241.88.88 0 01-1.242 0l-3.541-3.541v-.659l-.225-.233a5.417 5.417 0 01-4.45 1.233c-2.317-.391-4.167-2.325-4.45-4.658a5.42 5.42 0 016.058-6.058c2.334.283 4.267 2.133 4.659 4.45a5.417 5.417 0 01-1.234 4.45l.233.225zM4.565 8.31a3.745 3.745 0 003.75 3.75 3.745 3.745 0 003.75-3.75 3.745 3.745 0 00-3.75-3.75 3.745 3.745 0 00-3.75 3.75z"
        fill={
          Object {
            "payload": 4289574333,
            "type": 0,
          }
        }
        fillRule={0}
        propList={
          Array [
            "fill",
            "fillRule",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <TextInput
    onBlur={[Function]}
    onFocus={[Function]}
    placeholderTextColor="#868E96"
    returnKeyType="done"
    style={
      Array [
        Object {
          "color": "#0D131B",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontWeight": "400",
          "marginLeft": 10,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
    textAlignVertical="center"
  />
</View>
`;
